src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/CleanUpRejectedFilesTask.java

```diff
@@ -53,13 +56,22 @@ public void run() {
                     .listBlobs(null, true, EnumSet.of(SNAPSHOTS), null, null)
                     .forEach(listItem -> {
                         try {
+                            final String zipFileName = FilenameUtils.getName(listItem.getUri().toString());
                             CloudBlockBlob blob = container.getBlockBlobReference(
-                                FilenameUtils.getName(listItem.getUri().toString())
+                                zipFileName
                             );
 
                             if (canBeDeleted(blob)) {
-                                blob.delete(INCLUDE_SNAPSHOTS, null, null, null);
-                                log.info("Deleted rejected file {}", blob.getName());
+                                Optional<String> leaseId = blobManager.acquireLease(
+                                    blob,
+                                    container.getName(),
+                                    zipFileName
+                                );
+                                if (leaseId.isPresent()) {
+                                    AccessCondition accessCondition = generateLeaseCondition(leaseId.get());
+                                    blob.delete(INCLUDE_SNAPSHOTS, accessCondition, null, null);
```

Comment:
in v12 the delete function calls another one with all 4 arguments passed. last one being `Context.NONE`. is it same/similar with this one?

Reply:
The point is to use the access condition. CloudBlockBlob::delete(<no arguments>) calls internally delete(...) with 4 parameters with NULL access condition.